### ISSUE TITLE
Proposal: Markuplint MCP Server (Natural Language Rule Provider for AI Agents)

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#2616]
## Summary

To enable AI coding agents to correctly understand and follow Markuplint rules before generating code, I propose a new component: the Markuplint MCP Server.

This server reads Markuplint configuration files (e.g., .markuplintrc) and provides natural language descriptions in Markdown format for each rule. It acts as a human-/agent-readable layer between raw configuration and actual code generation.

### 📉 Lack of Embedding Coverage
- Current LLMs have very little embedded knowledge about Markuplint.
- Especially for custom rules or project-specific configurations, the models have no understanding.
- Rule names and JSON settings alone are insufficient for correct interpretation by AI agents.

### 🔄 Model Lag vs Rule Updates
- Markuplint rules are updated frequently.
- LLMs, however, are updated infrequently, leading to outdated knowledge.
- As a result, “good intentions” by the AI can violate current rules unknowingly.

### ❗ CI Feedback Comes Too Late
- Even though CI tools can catch rule violations, they only do so after the fact.
- For AI agents, what’s needed is proactive awareness, before the prompt is processed.
- This is not about catching errors—it’s about preventing them from happening.

### ✅ Function
- Reads .markuplintrc or related configuration files.
- Outputs Markdown

### 📄 Example Output (Markdown)

```md
- ✅️ Require the `alt` attribute in `img` element
- ✅️ Require an `h1` element in a document
```

## Use Cases

- 🔮 **Prompt-time rule guidance** for AI agents like Cursor or Cline Plugins.
- 📚 Clear, centralized **rule-sharing among team members**.
- 🧠 **Interactive rule explanations** inside editors like VS Code.
- 🔁 CI feedback with **natural language descriptions** (secondary use—not the primary goal).

## Implementation Sketch

- Simple API server (use `@modelcontextprotocol/sdk`).
- Input: repository path, config file, or rule name.
- Output: Markdown (single rule or full set).

## Notes

This proposal assumes that **AI doesn’t know everything** and **needs help understanding developer intent**.

By translating raw configuration into structured natural language, we can significantly improve the accuracy and safety of AI-assisted markup.
